src/shell/info.h

```diff
@@ -0,0 +1,49 @@
+/************************************************************\
+ * Copyright 2019 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _SHELL_INFO_H
+#define _SHELL_INFO_H
+
+#include <flux/core.h>
+#include <jansson.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "jobspec.h"
+#include "rcalc.h"
+
+struct shell_info {
+    flux_jobid_t jobid;
+    int shell_rank;
+    int shell_size;
```

Comment:
`shell_size` made me think "size of the shell", which makes no sense.  Should it really be `job_node_count` or something more like that?

Reply:
I struggled with that one!  I was trying to make it clear that the shells are themselves a parallel program of sorts with a rank and size.  Hmm, let me ponder some more.

Reply:
Er this was left as is in the current PR for now.  I'm open to changing later.